projects/pgai/tests/vectorizer/cli/test_vectorizer_document.py

```diff
@@ -0,0 +1,476 @@
+        result = run_vectorizer_worker(cli_db_url, vectorizer_id)
+
+    if result.exception:
+        print(result.stdout)
+
+    with connection.cursor(row_factory=dict_row) as cur:
+        cur.execute("SELECT count(*) FROM ai._vectorizer_q_1;")
+        assert cur.fetchone()["count"] == 1  # type: ignore
+        cur.execute("SELECT id, message, details FROM ai.vectorizer_errors")
+        records = cur.fetchall()
+        assert len(records) == 1
+        error = records[0]
+        assert error["id"] == vectorizer_id
+        assert error["message"] == "loading failed"
+        assert error["details"] == {
+            "loader": "uri",
+            "error_reason": f"unable to access bucket: '{s3_bucket}'"
+            " key: 'non_existing_doc.pdf' version: None"
+            " error: An error occurred (NoSuchKey) when"
+            " calling the GetObject operation: The specified"
+            " key does not exist.",
+            "is_retryable": True,
+        }
+
+        cur.execute(
+            "SELECT loading_retries, loading_retry_after" " FROM ai._vectorizer_q_1;"
+        )
+        queue_item = cur.fetchone()
+        assert queue_item["loading_retries"] == 1  # type: ignore
+        assert queue_item["loading_retry_after"] > datetime.now(tz=timezone.utc)  # type: ignore
+
+
+def test_there_will_be_no_more_retries_after_the_sixth_failure(
+    s3_bucket: str,
+    cli_db: tuple[TestDatabase, Connection],
+    cli_db_url: str,
+    vcr_: Any,
+):
+    """Test that embedding documents are successfully retried"""
+    connection = cli_db[1]
+    vectorizer_id = configure_document_vectorizer(
+        cli_db[1], base_path=f"s3://{s3_bucket}", documents=["non_existing_doc.pdf"]
+    )
+
+    with connection.cursor(row_factory=dict_row) as cur:
+        cur.execute(
+            "UPDATE ai._vectorizer_q_1"
+            " SET loading_retries=6, loading_retry_after=now() - interval '1 minute';"
+        )
+
+    with vcr_.use_cassette("doc_retries_s3_not_found.yaml"):
+        result = run_vectorizer_worker(cli_db_url, vectorizer_id)
+
+    if result.exception:
+        print(result.stdout)
+
+    with connection.cursor(row_factory=dict_row) as cur:
+        cur.execute("SELECT id, message, details FROM ai.vectorizer_errors")
+        records = cur.fetchall()
+        assert len(records) == 1
+        error = records[0]
+        assert error["id"] == vectorizer_id
+        assert error["message"] == "loading failed"
+        assert error["details"] == {
+            "loader": "uri",
+            "error_reason": f"unable to access bucket: '{s3_bucket}'"
+            " key: 'non_existing_doc.pdf' version: None"
+            " error: An error occurred (NoSuchKey) when"
+            " calling the GetObject operation: The specified"
+            " key does not exist.",
+            "is_retryable": False,
+        }
+
+        cur.execute(
+            "SELECT loading_retries, loading_retry_after" " FROM ai._vectorizer_q_1;"
+        )
```

Comment:
what does this do? the result is never checked

Reply:
fixed, thanks!